libpod/runtime_pod.go

```diff
@@ -25,6 +25,19 @@ type PodFilter func(*Pod) bool
 // being removed
 // Otherwise, the pod will not be removed if any containers are running
 func (r *Runtime) RemovePod(ctx context.Context, p *Pod, removeCtrs, force bool) error {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	if !r.valid {
+		return ErrRuntimeStopped
+	}
+
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	if !p.valid {
```

Comment:
Change this valid check to something more similar to https://github.com/projectatomic/libpod/blob/5d192d218b831d4bcd934e7b0a80a9e0586b4dfe/libpod/runtime_ctr.go#L203-L208

And move it back into removePod - we need this check even for internal calls